onnxruntime/core/framework/utils.h

```diff
@@ -41,6 +41,28 @@ namespace utils {
 void* DefaultAlloc(size_t size);
 void DefaultFree(void* p);
 
+/// <summary>
+// Do the placement new for strings on pre-allocated buffer
+// `elements` times.
+/// </summary>
+/// <param name="p_data"></param>
+/// <param name="elements"></param>
+void ConstructStrings(void* p_data, int64_t elements);
+
+/// <summary>
+/// The function uses an allocator (not null)
+/// and releases the supplied buffer. If is_string is true
+/// then the function assumes the buffer contains instances
+/// of std::string and calls ~string() on each instance `elements`
+/// times total.
+/// </summary>
+/// <param name="allocator"></param>
+/// <param name="is_string"></param>
+/// <param name="p_data"></param>
+/// <param name="elements"></param>
+void ReleaseTensorBuffer(const AllocatorPtr& allocator, bool is_string,
```

Comment:
>ReleaseTensorBuffer [](http://example.com/codeflow?start=5&length=19)

nit: how about DestroyStrings() instead for consistency?
also, if we can use C++17 (maybe from sparse_tensor.cc), how about these?
[https://en.cppreference.com/w/cpp/memory/uninitialized_value_construct_n](https://en.cppreference.com/w/cpp/memory/uninitialized_value_construct_n)
[https://en.cppreference.com/w/cpp/memory/destroy_n](https://en.cppreference.com/w/cpp/memory/destroy_n)

Reply:
This also deallocates the buffer contrary to ConstructStrings(). I can introduce a separate method for DestroyStrings() that is called from ReleaseBuffer(), would that work?

Reply:
sure, that works